src/Twig/Components/Search.php

```diff
@@ -3,64 +3,176 @@
 namespace App\Twig\Components;
 
 use ACSEO\TypesenseBundle\Finder\CollectionFinder;
-use ACSEO\TypesenseBundle\Finder\TypesenseQuery;
-use App\Entity\Book;
+use ACSEO\TypesenseBundle\Finder\TypesenseResponse;
+use App\Entity\Shelf;
+use App\Entity\User;
+use App\Search\QueryTokenizer;
+use App\Search\TypesenseTokenHandler;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\SecurityBundle\Security;
 use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
 use Symfony\UX\LiveComponent\Attribute\LiveAction;
 use Symfony\UX\LiveComponent\Attribute\LiveArg;
 use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\LiveComponent\ComponentToolsTrait;
 use Symfony\UX\LiveComponent\DefaultActionTrait;
+use Symfony\UX\TwigComponent\Attribute\PostMount;
 
 #[AsLiveComponent(method: 'get')]
 class Search
 {
     use DefaultActionTrait;
+    use ComponentToolsTrait;
 
-    #[LiveProp(writable: true, url: true)]
+    public const PER_PAGE = 18;
+
+    #[LiveProp(writable: true)]
     public string $query = '';
     public array $books = [];
+    public array $facets = [];
+    public array $suggestions = [];
+    public int $found = 0;
+
+    #[LiveProp(writable: true)]
+    public array $filters = [];
+
+    #[LiveProp(writable: true, url: true)]
+    public string $fullQuery = '';
+
+    #[LiveProp(writable: true, url: true)]
+    public int $page = 1;
+
+    #[LiveProp(writable: true)]
+    public string $shelfname = '';
+
+    public array $pagination = [];
 
-    public function __construct(protected CollectionFinder $bookFinder)
+    public function __construct(protected CollectionFinder $bookFinder, protected QueryTokenizer $queryTokenizer, protected TypesenseTokenHandler $tokenHandler, protected Security $security, protected EntityManagerInterface $manager)
     {
     }
 
+    #[PostMount]
+    public function postMount(): void
+    {
+        $tokens = $this->extractFilters(true);
+        $this->getResults($tokens);
+    }
+
     #[LiveAction]
     public function addToQuery(#[LiveArg] string $value): void
```

Comment:
This method do way more than `addToQuery` (alter the filters, extract token, etc), "adding" feels confusing,  rename to something else like `alterQuery`